pkg/sentry/socket/hostinet/socket_unsafe.go

```diff
@@ -69,6 +69,38 @@ func (s *socketOperations) Ioctl(ctx context.Context, _ *fs.File, io usermem.IO,
 		})
 		return 0, err
 
+	case syscall.SIOCGIFFLAGS,
+		syscall.SIOCGIFADDR,
+		syscall.SIOCGIFBRDADDR,
+		syscall.SIOCGIFDSTADDR,
+		syscall.SIOCGIFHWADDR,
+		syscall.SIOCGIFINDEX,
+		syscall.SIOCGIFMAP,
+		syscall.SIOCGIFMETRIC,
+		syscall.SIOCGIFMTU,
+		syscall.SIOCGIFNAME,
+		syscall.SIOCGIFNETMASK,
+		syscall.SIOCGIFTXQLEN:
+		var ifr linux.IFReq
+		if _, err := usermem.CopyObjectIn(ctx, io, args[2].Pointer(), &ifr, usermem.IOOpts{
+			AddressSpaceActive: true,
+		}); err != nil {
+			return 0, err
+		}
+
+		if _, err := socket.InterfaceIoctl(ctx, io, int(cmd), &ifr); err != nil {
+			return 0, err.ToError()
+		}
+
+		_, err := usermem.CopyObjectOut(ctx, io, args[2].Pointer(), &ifr, usermem.IOOpts{
+			AddressSpaceActive: true,
+		})
+
+		return 0, err
+
+	case syscall.SIOCGIFCONF:
+		return 0, socket.IfconfIoctl(ctx, io, args[2].Pointer())
+
```

Comment:
I understand why these ioctls might be useful for some users, but it's too close to genuine passthrough for gVisor. Enabling these ioctls -- network interface names, addresses, etc. -- leaks too much information about the host and gives applications access to a larger chuck of the host syscall interface.

We talked and would prefer to accept everything in this patch _except_ the changes here to `socket_unsafe.go`. That way it's easy to maintain a diff if you choose to do so.

What do you think?

Reply:
As I understand, hostinet is always not targeting a safe network solution. The other side, it's a much more stable one than others as it just reuses host tcp/ip stack.

So we value usability than security here?

> Enabling these ioctls -- network interface names, addresses, etc. -- leaks too much information about the host

It indeed exposes information. But such information can be easily got from other ways, say getsockname/netlink.

> and gives applications access to a larger chuck of the host syscall interface.

The syscall interface (seccomp) to host is not expanded in this patchset. The security basis here is not based on information hide, but the seccomp whitelist?

Reply:
> On the other hand, I think @kevinGC has a good point that this leaks information about the host system

It leaks the network information for this container netns, instead of the whole host domain netns. And as you mentioned, this information is already assumed to be exposed if sentry is compromised. As again, this information is already exposed to user by netlink -> stack.Interfaces, no?

Reply:
Plus, netstack also learns the network information from container netns. And it also exposes the information to user applications.

I guess when you say exposing information, do you mean the case when --network=host is specified? In that case, we indeed expose all network information on the host. But --network=host is not a secure way to use in itself.

Reply:
@kevinGC, any additional thoughts on this matter?

Reply:
Given that a) this  is for hostinet and b) only exposes the nic's inside the namespace it should be fine to make this a pass through.